src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -328,15 +329,40 @@ private void AttachToDocument(uint docCookie, string moniker)
             // This should always succeed since we only got here if we already confirmed the moniker is acceptable
             var languageInformation = TryGetLanguageInformation(moniker);
             Contract.ThrowIfNull(languageInformation);
-            var parseOptions = languageInformation.ParseOptions;
 
-            if (Path.GetExtension(moniker) == languageInformation.ScriptExtension)
+            var languageServices = Services.GetLanguageServices(languageInformation.LanguageName);
+            var compilationFactory = languageServices.GetService<ICompilationFactoryService>();
+            var compilationOptions = compilationFactory.GetDefaultCompilationOptions();
+            var syntaxTreeFactory = languageServices.GetService<ISyntaxTreeFactoryService>();
+            var parseOptions = syntaxTreeFactory.GetDefaultParseOptions();
+
+            if (PathUtilities.GetExtension(moniker) == languageInformation.ScriptExtension)
             {
                 parseOptions = parseOptions.WithKind(SourceCodeKind.Script);
```

Comment:
VB CompilationOptions also contain the ParseOptions, so I suspect you need to update CompilationOptions too, or leave a comment why the stuff that is impacted by that scenario doesn't matter here. I suspect it doesn't, but the comment would be good.